### PR TITLE
Fix LFM2-MoE sigmoid routing and expert-bias selection

### PR DESCRIPTION
**Pull request by ykhrustalev**
## Proposed changes

Fix LFM2-MoE routing to match the sigmoid-gated behavior used by LFM2.5 checkpoints.

The current Swift path applies softmax to router logits and folds `expert_bias` into the combination weights. LFM2-MoE uses per-expert sigmoid scores: `expert_bias` should affect top-k selection only, while the returned weights stay gathered from the unbiased sigmoid scores, optionally normalized by `norm_topk_prob`, then scaled by `routed_scaling_factor`.

This mirrors the same routing fix in ml-explore/mlx-lm#1354.

This also adds focused routing regression tests for the sigmoid gate, expert-bias selection-only behavior, and top-k normalization.

The dense LFM2-MoE MLP path now compiles the SwiGLU activation product, matching the existing compiled SwitchGLU path used by the MoE experts.

## Checklist

Put an `x` in the boxes that apply.

- [x] I have read the [CONTRIBUTING](https://github.com/ml-explore/mlx/blob/main/CONTRIBUTING.md) document
- [x] I have run `pre-commit run --all-files` to format my code / installed pre-commit prior to committing changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the necessary documentation (if needed)

### CODE REVIEW

**Comment by ykhrustalev** [comment]
cc @nastya236

File: Libraries/MLXLLM/Models/LFM2MoE.swift [hunk 1]
```diff
@@ -12,6 +12,12 @@ import MLX
 import MLXLMCommon
 import MLXNN
 
+private let compiledLFM2MoESiluProduct: @Sendable (MLXArray, MLXArray) -> MLXArray = compile(
```
Comment (davidkoski) [inline Libraries/MLXLLM/Models/LFM2MoE.swift]: Let's use this:

```swift
private let compiledSiluProduct: @Sendable (MLXArray, MLXArray) -> MLXArray = ...
```

from SwichLayers.swift -- identical function.  Promote that to public (like the `weightedExpertSum` next to it).

Other than this small item, looks good!

**Review by davidkoski — APPROVED** [review]
Thank you!
